source/core_http_client.c

```diff
@@ -1111,6 +1129,10 @@ static HTTPStatus_t parseHttpResponse( HTTPParsingContext_t * pParsingContext,
                                        pParsingContext->pBufferCur,
                                        parseLen );
 
+    /* The next location to parse will always be after what has already
+     * been parsed. */
+    pParsingContext->pBufferCur = parsingStartLoc + bytesParsed;
```

Comment:
Would it be good practice to assert that?

Reply:
I am explaining here why we  move pBufferCur forward. I don't see where to assert.